frontend/lib/src/components/shared/ElementFullscreen/useFullscreen.tsx

```diff
@@ -0,0 +1,79 @@
+/**
+ * Copyright (c) Streamlit Inc. (2018-2022) Snowflake Inc. (2022-2024)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { useCallback, useContext, useEffect, useMemo, useState } from "react"
+
+import { LibContext } from "@streamlit/lib/src/components/core/LibContext"
+import { WindowDimensionsContext } from "@streamlit/lib/src/components/shared/WindowDimensions"
+import { useRequiredContext } from "@streamlit/lib/src/hooks/useRequiredContext"
+
+export type UseEscapeToCollapseArgs = {
+  expanded: boolean
+  zoomOut: () => void
+}
+
+export type UseFullscreenShape = {
+  expanded: boolean
+  fullHeight: number
+  fullWidth: number
+  zoomIn: () => void
+  zoomOut: () => void
+}
+
+export const useFullscreen = (): UseFullscreenShape => {
+  const libContext = useContext(LibContext)
+  const [expanded, setExpanded] = useState(false)
+  const { fullHeight, fullWidth } = useRequiredContext(WindowDimensionsContext)
+  const { setFullScreen } = libContext
```

Comment:
supernit: if we aren't using `libContext` elsewhere, we can just destructure `setFullscreen` on line 37 🤷